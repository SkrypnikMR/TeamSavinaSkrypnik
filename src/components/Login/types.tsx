import React from 'react';

import { ITheme } from '../App/types';

export interface IDiv extends ITheme{
    display: string;
    justifyContent: string;
    alignItems: string;
    background: string;
    height: string;
}
