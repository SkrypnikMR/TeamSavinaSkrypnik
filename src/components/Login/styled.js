import styled from 'styled-components';

export const StLoginDiv = styled.div`
    display: flex;
    justify-content: center;
    align-items: center;
`;
