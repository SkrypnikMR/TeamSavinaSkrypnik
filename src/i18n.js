import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';

import en from './locales/en.json';
import ar from './locales/ar.json';
import ru from './locales/ru.json';

const language = localStorage.getItem('lang') || 'en';

i18n.use(initReactI18next).init({
    resources: {
        en: { translations: en },
        ar: { translations: ar },
        ru: { translations: ru },
    },
    fallbackLng: ['en', 'ar', 'ru'],
    lng: language,
    debug: false,
    ns: ['translations'],
    defaultNS: 'translations',
    keySeparator: false,
    interpolation: {
        escapeValue: false,
        formatSeparator: ',',
    },
    react: {
        wait: true,
    },
});

export default i18n;
