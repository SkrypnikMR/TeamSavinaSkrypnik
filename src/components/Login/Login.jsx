import React from 'react';
import PropTypes from 'prop-types';
import { logInputs } from '/src/constants/componentsÐ¡onsts';
import { NavLink, Redirect } from 'react-router-dom';
import { useTranslation } from 'react-i18next';
import { APP_ROUTES } from '/src/constants/reactRoutes';
import { StLoginDiv } from './styled';
import Form from '../UI/Form';
import Input from '../UI/Input';
import Button from '../UI/Button';

const Login = ({ sendLoginRequest, setLoginValue, fields, setValue }) => {
  const { t } = useTranslation();

  if (fields.success) {
    setValue({ name: 'success', value: false });
    return <Redirect to={APP_ROUTES.chat} />;
  }
  return (
    <StLoginDiv>
      <Form>
        <p>{t('authorization')}</p>
        {logInputs.map(input => (
          <Input
            width="80%"
            id={input.id}
            height="100px"
            key={input.id}
            name={input.id}
            inputHeight="50px"
            borderRadius="5px"
            label={t(input.label)}
            value={fields[input.id]}
            onChange={setLoginValue}
            placeholder={t(input.placeholder)}
          />
        ))}
        <Button
          height="50px"
          content="login_btn"
          id="login"
          onClick={sendLoginRequest}
        />
        <span>
          {t('login_span_text')}
          <NavLink to={APP_ROUTES.registration}>{t('login_link_text')}</NavLink>
        </span>
      </Form>
    </StLoginDiv>
  );
};

Login.propTypes = {
  sendLoginRequest: PropTypes.func.isRequired,
  setLoginValue: PropTypes.func.isRequired,
  fields: PropTypes.object.isRequired,
};

export default Login;
