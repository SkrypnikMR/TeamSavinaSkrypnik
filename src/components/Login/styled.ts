import styled from 'styled-components';

import { IDiv } from './types';

export const StLoginDiv = styled.div < IDiv > `
    display: flex;
    justify-content: center;
    align-items: center;
    background: ${({ colors, theme }) => colors[theme].backgroundLogReg};
    height: 100vh;
    img{
        height: 40vh;
    }
`;
