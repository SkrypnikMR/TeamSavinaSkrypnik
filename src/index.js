import React from 'react';
import { render } from 'react-dom';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from '@redux-saga/core';
import rootReducer from './Store';
import App from './components/App';
import rootSaga from './Store/saga';
import './i18n';

const devTools = window?.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__();
const saga = createSagaMiddleware();

const store = createStore(
  rootReducer,
  compose(
    applyMiddleware(saga),
    devTools,
  ),
);
saga.run(rootSaga);

render(
  <Provider store={store}>
    <App />
  </Provider >,
  document.getElementById('root'),
);
