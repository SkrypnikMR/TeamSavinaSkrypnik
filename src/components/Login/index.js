import { connect } from 'react-redux';
import Login from './Login.jsx';
import { setLoginValue, sendLoginRequest, setValue } from '/src/Store/login/actions';
import { loginStore } from '/src/Store/login/selectors';

const mapStateToProps = state => ({ fields: loginStore(state) });

const mapDispatchToProps = dispatch => ({
    setLoginValue: payload => dispatch(setLoginValue(payload)),
    sendLoginRequest: () => dispatch(sendLoginRequest()),
    setValue: payload => dispatch(setValue(payload)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Login);
