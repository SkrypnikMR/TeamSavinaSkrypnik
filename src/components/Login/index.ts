import { connect } from 'react-redux';
import { loginStore } from 'src/Store/login/selectors';
import { setLoginValue, sendLoginRequest, setValue } from 'src/Store/login/actions';
import { Dispatch } from 'redux';
import Login from './Login.jsx';
import { ApplicationState } from '../../store/types';
import { TLoginPayload } from '../../store/login/types';

const mapStateToProps = (state: ApplicationState) => ({ fields: loginStore(state) });

const mapDispatchToProps = (dispatch: Dispatch) => ({
    setLoginValue: (payload: TLoginPayload) => dispatch(setLoginValue(payload)),
    sendLoginRequest: () => dispatch(sendLoginRequest()),
    setValue: (payload: TLoginPayload) => dispatch(setValue(payload)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Login);
