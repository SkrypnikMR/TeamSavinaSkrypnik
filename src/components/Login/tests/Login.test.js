import React from 'react';
import { shallowSmart, mountSmart } from '/src/helpers/testHelper';
import Login from '../Login.jsx';
import '/src/i18n';

describe('Login', () => {
    let props;
    const sendLoginRequest = jest.fn();
    const setLoginValue = jest.fn();
    const fields = {
        email: '',
        password: '',
        success: null,
    };
    beforeEach(() => {
        props = {
            setLoginValue,
            sendLoginRequest,
            fields,
        };
    });
    it('Should match snapshot', () => {
        const component = shallowSmart(<Login {...props} />);
        expect(component.html()).toMatchSnapshot();
    });
    it('should render p', () => {
        const component = mountSmart(<Login {...props} />);
        expect(component.find('p')).toHaveLength(1);
    });
    it('should render inputs', () => {
        const component = mountSmart(<Login {...props} />);
        expect(component.find('Input')).toHaveLength(2);
    });
    it('should render button', () => {
        const component = mountSmart(<Login {...props} />);
        expect(component.find('Button')).toHaveLength(1);
    });
    it('should click on the button', () => {
        const component = mountSmart(<Login {...props} />);
        component.find('Button').simulate('click');
        expect(sendLoginRequest).toHaveBeenCalled();
    });
    it('should click change input', () => {
        const component = mountSmart(<Login {...props} />);
        component.find('input').first().simulate('change', { target: { name: 'email', value: 'emailValue' } });
        expect(setLoginValue).toHaveBeenCalledWith({ name: 'email', value: 'emailValue' });
    });
    it('should redirect', () => {
        props.fields = {
            success: true,
        };
        const component = mountSmart(<Login {...props} />);
        expect(component.find('Redirect')).toHaveLength(1);
    });
});
