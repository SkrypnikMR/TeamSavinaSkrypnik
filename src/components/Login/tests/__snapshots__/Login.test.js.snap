// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Should match snapshot 1`] = `"<div class=\\"styled__StLoginDiv-ohf2t6-0 cCvQFI\\"><div class=\\"styled__StFormDiv-sc-1xa63f-0 hNLlQX\\"><p>Sign In</p><div width=\\"80%\\" height=\\"100px\\" class=\\"styled__StInputContainer-w3ykst-3 emTECj\\"><label for=\\"email\\" class=\\"styled__StLabel-w3ykst-0 fbdCHZ\\">Email</label><input type=\\"text\\" id=\\"email\\" name=\\"email\\" value=\\"\\" placeholder=\\"Enter your email...\\" class=\\"styled__StInput-w3ykst-1 eAZEdV\\"/></div><div width=\\"80%\\" height=\\"100px\\" class=\\"styled__StInputContainer-w3ykst-3 emTECj\\"><label for=\\"password\\" class=\\"styled__StLabel-w3ykst-0 fbdCHZ\\">Password</label><input type=\\"text\\" id=\\"password\\" name=\\"password\\" value=\\"\\" placeholder=\\"Enter your password...\\" class=\\"styled__StInput-w3ykst-1 eAZEdV\\"/></div><button id=\\"login\\" type=\\"button\\" height=\\"50px\\" class=\\"styled__StButton-sc-1a6hck8-0 euTlOK\\">Login</button><span>Need an account?<a href=\\"/registration\\">Register now</a></span></div></div>"`;
